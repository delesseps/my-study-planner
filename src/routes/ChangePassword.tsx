import React, {useEffect} from 'react'
import {AxiosError} from 'axios'
import {LockOutlined} from '@ant-design/icons'
import {Button, Input, message, Form} from 'antd'
import {Link, match, useHistory} from 'react-router-dom'
import styled from 'styled-components'

import {ReactComponent as Logo} from 'assets/logo.svg'
import {ReactComponent as ChangePasswordDone} from 'assets/change_password_done.svg'
import {FadeIn, Loading} from 'components'
import {usePasswordChange} from 'features/auth/auth-context'

interface IChangePasswordProps {
  match: match<{email: string; token: string}>
}

const ChangePassword: React.FunctionComponent<IChangePasswordProps> = ({
  match,
}) => {
  const [form] = Form.useForm()
  const {push} = useHistory()
  const {
    confirmToken: [
      confirmToken,
      {status: tokenConfirmationStatus, error: tokenConfirmationError},
    ],
    change: [
      changePassword,
      {status: passwordChangeStatus, error: passwordChangeError},
    ],
  } = usePasswordChange()
  const [confirmDirty, setConfirmDirty] = React.useState<Boolean | any>(false)

  useEffect(() => {
    confirmToken(match.params.token)
  }, [match.params.token, confirmToken])

  React.useEffect(() => {
    const error = tokenConfirmationError as AxiosError
    const errorCode = error?.response?.status

    if (errorCode) {
      message.error('Invalid link')
      push('/forgot_password')
    }

    if (!errorCode && error) {
      throw new Error(error.toString())
    }
  }, [tokenConfirmationError, push])

  React.useEffect(() => {
    const error = passwordChangeError as AxiosError
    const errorCode = error?.response?.status

    if (errorCode) {
      message.error('An error has occured please try again!')
      push('/forgot_password')
    }

    if (!errorCode && error) {
      throw new Error(error.toString())
    }
  }, [passwordChangeError, push])

  const handleSubmit = (): void => {
    form.validateFields().then(credentials => SendRequest(credentials.password))
  }

  const SendRequest = async (password: string) => {
    const {token} = match.params
    changePassword({token, password})
  }

  const handleConfirmBlur = (e: React.BaseSyntheticEvent) => {
    const value = e.target.value
    setConfirmDirty({confirmDirty: confirmDirty || !!value})
  }

  const compareToFirstPassword = (
    rule: any,
    value: string,
    callback: Function,
  ) => {
    if (value && value !== form.getFieldValue('password')) {
      callback('The passwords do not match!')
    } else {
      callback()
    }
  }

  const validateToNextPassword = (
    rule: any,
    value: string,
    callback: Function,
  ) => {
    if (value && confirmDirty) {
      form.validateFields(['confirm'])
    }
    callback()
  }

  const success = passwordChangeStatus === 'success'
  const passwordChangeLoading = success || passwordChangeStatus === 'loading'

  if (tokenConfirmationStatus === 'loading') {
    return <Loading />
  } else {
    return (
      <FadeIn>
        <Wrapper>
          <LogoBox>
            <StyledLogo />
            <Title>My Study Planner</Title>
          </LogoBox>
          <Card>
            <CardTitle>Change Password</CardTitle>
            <CardBody>
              {success
                ? 'Successfuly changed password!'
                : 'Please enter your new password.'}
            </CardBody>
            {success ? (
              <ChangePasswordDone />
            ) : (
              <StyledForm form={form} layout="vertical" onFinish={handleSubmit}>
                <Form.Item
                  name="password"
                  rules={[
                    {
                      required: true,
                      message: 'Please input your password!',
                    },
                    {
                      validator: validateToNextPassword,
                    },
                    {
                      min: 6,
                      message: 'Password must have a minimum of 6 characters.',
                    },
                  ]}
                  hasFeedback
                >
                  <Input.Password
                    prefix={<LockOutlined style={{color: 'rgba(0,0,0,.25)'}} />}
                    type="password"
                    placeholder="New password"
                  />
                </Form.Item>
                <Form.Item
                  name="confirm"
                  rules={[
                    {
                      required: true,
                      message: 'Please confirm your password!',
                    },
                    {
                      validator: compareToFirstPassword,
                    },
                    {
                      min: 6,
                      message: 'Password must have a minimum of 6 characters.',
                    },
                  ]}
                  hasFeedback
                >
                  <Input.Password
                    prefix={<LockOutlined style={{color: 'rgba(0,0,0,.25)'}} />}
                    type="password"
                    placeholder="Confirm new password"
                    onBlur={handleConfirmBlur}
                  />
                </Form.Item>
                <Form.Item>
                  <Button
                    loading={passwordChangeLoading}
                    block
                    type="primary"
                    size="large"
                    htmlType="submit"
                  >
                    Reset Password
                  </Button>
                </Form.Item>
              </StyledForm>
            )}

            <StyledLink to="/signin">Sign in</StyledLink>
          </Card>
        </Wrapper>
      </FadeIn>
    )
  }
}

const Wrapper = styled.main`
  display: flex;
  flex-direction: column;
  align-items: center;

  padding: 4rem 0;
  min-height: 100vh;

  background-color: ${props => props.theme.colors.main};
`

const LogoBox = styled.div`
  display: flex;
  align-items: center;

  margin-bottom: 8rem;
`

const StyledLogo = styled(Logo)`
  width: 5rem;
  height: 5rem;

  fill: white;

  margin-right: 1rem;
`

const Title = styled.h1`
  font-weight: 600;
  margin-bottom: 0;
  color: white;
`

const Card = styled.div`
  background-color: #fff;

  display: flex;
  flex-direction: column;
  align-items: center;

  padding: 2rem 4rem;

  width: 44rem;

  border-radius: 6px;

  box-shadow: 0 1rem 2rem rgba(0, 0, 0, 0.2);

  & > *:not(:last-child) {
    margin-bottom: 2rem;
  }
`

const CardTitle = styled.h2`
  font-weight: 300;
  font-size: 2.4rem;
  color: ${props => props.theme.colors.main};
`

const CardBody = styled.h4`
  font-weight: 400;
  color: rgba(39, 39, 39, 0.7);

  && {
    margin-bottom: 3rem;
  }
`

const StyledForm = styled(Form)`
  width: 100%;
  && {
    margin-bottom: 0;
  }
`

const StyledLink = styled(Link)`
  font-size: 1.6rem;
`

export default ChangePassword
