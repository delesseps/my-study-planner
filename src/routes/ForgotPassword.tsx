import * as React from 'react'
import styled from 'styled-components'
import {UserOutlined} from '@ant-design/icons'
import {Button, Input, Alert, Form} from 'antd'
import {Link} from 'react-router-dom'

import {FadeIn} from 'components'
import {usePasswordChange} from 'features/auth/auth-context'
import {ReactComponent as Logo} from 'assets/logo.svg'
import {ReactComponent as SentMessage} from 'assets/message_sent.svg'
import {AxiosError} from 'axios'

const ForgotPassword: React.FunctionComponent = () => {
  const [form] = Form.useForm()
  const {
    request: [requestChange, {status, error}],
  } = usePasswordChange()

  const handleSubmit = (): void => {
    form.validateFields().then(value => SendRequest(value.email))
  }

  const SendRequest = async (email: string) => {
    requestChange(email)
  }

  const success = status === 'success'
  const errorObject = error as AxiosError
  const errorCode = errorObject?.response?.status

  return (
    <FadeIn>
      <Wrapper>
        <LogoBox>
          <StyledLogo />
          <Title>My Study Planner</Title>
        </LogoBox>
        {errorCode && errorCode !== 400 && (
          <StyledError
            message="Error sending recovery link, is this the right email?"
            type="error"
            showIcon
          />
        )}
        {errorCode === 400 && (
          <StyledError
            message="Google linked accounts can not recover a password!"
            type="error"
            showIcon
          />
        )}
        <Card>
          <CardTitle>Reset Password</CardTitle>
          <CardBody>
            {success
              ? 'Successfuly sent! Please check your email!'
              : "Please enter your email address below and we'll send you a link to reset your password"}
          </CardBody>
          {success ? (
            <SentMessage />
          ) : (
            <StyledForm form={form} layout="vertical" onFinish={handleSubmit}>
              <Form.Item
                name="email"
                rules={[
                  {required: true, message: 'Please input your email!'},
                  {
                    type: 'email',
                    message: 'The input is not valid E-mail!',
                  },
                ]}
              >
                <Input
                  prefix={<UserOutlined style={{color: 'rgba(0,0,0,.25)'}} />}
                  placeholder="Email address"
                />
              </Form.Item>
              <Form.Item>
                <Button
                  loading={status === 'loading' || status === 'success'}
                  block
                  type="primary"
                  size="large"
                  htmlType="submit"
                >
                  Reset Password
                </Button>
              </Form.Item>
            </StyledForm>
          )}
          <LinkWrapper>
            <StyledLink to="/signin">Sign in</StyledLink>
            <h2>or</h2>
            <StyledLink to="/signup">Sign up</StyledLink>
          </LinkWrapper>
        </Card>
      </Wrapper>
    </FadeIn>
  )
}

const Wrapper = styled.main`
  display: flex;
  flex-direction: column;
  align-items: center;

  padding: 4rem 0;

  min-height: 100vh;

  background-color: ${props => props.theme.colors.main};
`

const LogoBox = styled.div`
  display: flex;
  align-items: center;

  margin-bottom: 8rem;
`

const StyledLogo = styled(Logo)`
  width: 5rem;
  height: 5rem;

  fill: white;

  margin-right: 1rem;
`

const Title = styled.h1`
  font-weight: 600;
  margin-bottom: 0;
  color: white;
`

const StyledError = styled(Alert)`
  margin-bottom: 1rem;
`

const Card = styled.div`
  background-color: #fff;

  display: flex;
  flex-direction: column;
  align-items: center;

  padding: 2rem 4rem;

  width: 44rem;

  border-radius: 6px;

  box-shadow: 0 1rem 2rem rgba(0, 0, 0, 0.2);

  & > *:not(:last-child) {
    margin-bottom: 2rem;
  }
`

const CardTitle = styled.h2`
  font-weight: 300;
  font-size: 2.4rem;
  color: ${props => props.theme.colors.main};
`

const CardBody = styled.h4`
  font-weight: 400;
  color: rgba(39, 39, 39, 0.7);

  && {
    margin-bottom: 3rem;
  }
`

const StyledForm = styled(Form)`
  width: 100%;
  && {
    margin-bottom: 0;
  }
`

const LinkWrapper = styled.div`
  display: flex;
  align-items: center;
  flex: 1;

  & h2 {
    color: rgba(39, 39, 39, 0.6);
    margin-bottom: 0;

    font-size: 1.5rem;
  }

  & > *:not(:last-child) {
    margin-right: 1rem;
  }
`

const StyledLink = styled(Link)`
  font-size: 1.6rem;
`

export default ForgotPassword
