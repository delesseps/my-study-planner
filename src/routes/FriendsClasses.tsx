import React from 'react'

const FriendsClasses: React.FC = props => {
  return <div>Friends and classes</div>
}

export default FriendsClasses
