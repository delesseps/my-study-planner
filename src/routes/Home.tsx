import React from 'react'
import styled from 'styled-components'
import {useWindowSize} from 'react-use'

import {FadeIn, Profiler} from 'components'
import {breakpoints} from 'theme'
import {
  CounterPanel,
  EvaluationPanel,
  HomeworkPanel,
  ToDoPanel,
  RecommendedActionsPanel,
  CalendarPanel,
} from 'components/panels'
import {useAuth} from 'features/auth/auth-context'

const HomeSchedulePanel = React.lazy(() =>
  import('../components/panels/HomeSchedulePanel'),
)
const WelcomeModal = React.lazy(() => import('../components/modals/Welcome'))

const Home: React.FC = () => {
  const {user} = useAuth()
  const {width} = useWindowSize()

  const isMobile = width <= 425

  return (
    <Profiler id="Home Screen">
      <FadeIn>
        <Wrapper>
          <WelcomeModal />
          <CounterPanelLeft>
            <CounterPanel user={user} homework />
          </CounterPanelLeft>
          <CounterPanelRight>
            <CounterPanel user={user} />
          </CounterPanelRight>

          {!isMobile && (
            <HomeSchedulePanelWrapper>
              <HomeSchedulePanel />
            </HomeSchedulePanelWrapper>
          )}

          <ToDoPanelWrapper>
            <ToDoPanel />
          </ToDoPanelWrapper>

          <HomeworkPanelWrapper>
            <HomeworkPanel />
          </HomeworkPanelWrapper>

          <EvaluationPanelWrapper>
            <EvaluationPanel />
          </EvaluationPanelWrapper>

          <RecommendedActionsPanelWrapper>
            <RecommendedActionsPanel />
          </RecommendedActionsPanelWrapper>

          <CalendarPanelWrapper>
            <CalendarPanel />
          </CalendarPanelWrapper>
        </Wrapper>
      </FadeIn>
    </Profiler>
  )
}

const panelTheme = `border-radius: 4px;`

const Wrapper = styled.section`
  display: grid;
  margin-top: 6rem;
  grid-gap: 3rem;
  grid-template-areas:
    'left right ractions'
    'schedule schedule calendar'
    'todo homework evaluation ';
  grid-template-columns: repeat(3, 0.5fr);

  @media only screen and (max-width: ${breakpoints.bpLargeMedium}) {
    grid-template-areas:
      'left right'
      'schedule schedule'
      'evaluation ractions'
      'homework todo'
      'calendar calendar';
    grid-template-columns: repeat(2, 1fr);
  }

  @media only screen and (max-width: ${breakpoints.bpMobileL}) {
    grid-template-areas:
      'left '
      'right'
      'calendar'
      'evaluation'
      'homework'
      'ractions'
      'todo'
      'schedule';
    grid-template-columns: 1fr;
  }
`

const CounterPanelLeft = styled.div`
  box-shadow: ${props => props.theme.shadow1};
  height: 13rem;
  position: relative;

  padding: 2rem 2.6rem;
  background-color: ${props => props.theme.panelBackgroundColor};

  ${panelTheme};

  grid-area: left;
`

const CounterPanelRight = styled.div`
  ${panelTheme};
  box-shadow: ${props => props.theme.shadow1};
  height: 13rem;
  position: relative;
  background-color: ${props => props.theme.panelBackgroundColor};

  padding: 2rem 2.6rem;

  grid-area: right;
`

const CalendarPanelWrapper = styled.div`
  align-self: start;

  background-color: #fff;
  border-radius: 4px;
  box-shadow: ${props => props.theme.shadow1};
  background-color: ${props => props.theme.panelBackgroundColor};

  grid-area: calendar;
  grid-row: auto;
`

const HomeSchedulePanelWrapper = styled.div`
  height: 48.9rem;
  margin-top: -17rem;
  padding: 2rem 2.6rem;
  background-color: ${props => props.theme.panelBackgroundColor};

  grid-area: schedule;

  ${panelTheme};
  box-shadow: ${props => props.theme.shadow1};
  padding: 1rem;

  @media only screen and (max-width: ${breakpoints.bpLargest}) {
    height: 56.9rem;
  }

  @media only screen and (max-width: ${breakpoints.bpLargeMedium}) {
    margin-top: 0;
  }
`

const RecommendedActionsPanelWrapper = styled.div`
  ${panelTheme};
  box-shadow: ${props => props.theme.shadow1};
  height: 30rem;
  overflow-y: auto;
  background-color: ${props => props.theme.panelBackgroundColor};

  display: flex;
  flex-direction: column;

  grid-area: ractions;
  grid-row: auto;
`

const ToDoPanelWrapper = styled.div`
  ${panelTheme};

  display: flex;
  flex-direction: column;
  background-color: ${props => props.theme.panelBackgroundColor};

  box-shadow: ${props => props.theme.shadow1};
  min-height: 30rem;
  max-height: 60rem;

  overflow-y: auto;

  grid-area: todo;
`

const HomeworkPanelWrapper = styled.div`
  ${panelTheme};

  display: flex;
  flex-direction: column;
  background-color: ${props => props.theme.panelBackgroundColor};

  box-shadow: ${props => props.theme.shadow1};
  min-height: 30rem;
  max-height: 60rem;

  overflow-y: auto;

  grid-area: homework;
`

const EvaluationPanelWrapper = styled.div`
  ${panelTheme};

  display: flex;
  flex-direction: column;
  background-color: ${props => props.theme.panelBackgroundColor};

  box-shadow: ${props => props.theme.shadow1};
  min-height: 30rem;
  max-height: 60rem;

  overflow-y: auto;

  grid-area: evaluation;
`

export default Home
