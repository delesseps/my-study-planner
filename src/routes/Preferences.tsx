import React from "react";
import styled from "styled-components";
import { List, Switch } from "antd";
import { ApplicationState } from "store/types";
import IUserConfig from "interfaces/IUserConfig";
import { useDispatch, connect } from "react-redux";
import { updateUserConfig } from "store/effects";
import { breakpoints } from "styled";

interface IPreferencesProps {
  config: IUserConfig;
}

const appereanceSettings = [
  {
    name: "Dark Mode",
    description: "Make My Study Planner cooler for your eyes.",
    action: "DARK_MODE",
    configProperty: "darkMode"
  }
];

const Preferences: React.FunctionComponent<IPreferencesProps> = ({
  config
}) => {
  const dispatch = useDispatch();

  const handleChange = (action: string) => (
    checked: boolean,
    event: MouseEvent
  ) => {
    const newConfig: IUserConfig = config;

    switch (action) {
      case "DARK_MODE":
        newConfig.darkMode = checked;
        return dispatch(updateUserConfig(newConfig));
      default:
    }
  };

  return (
    <Wrapper>
      <Title>Appereance</Title>
      <StyledList
        bordered
        itemLayout="horizontal"
        dataSource={appereanceSettings}
        renderItem={(setting: any) => (
          <List.Item
            actions={[
              <Switch
                checked={config[setting.configProperty]}
                onChange={handleChange(setting.action)}
              />
            ]}
          >
            <List.Item.Meta
              title={<Name>{setting.name}</Name>}
              description={<Description>{setting.description}</Description>}
            />
          </List.Item>
        )}
      />
    </Wrapper>
  );
};

const Wrapper = styled.section`
  display: flex;
  flex-direction: column;

  margin: 0 auto;
  width: 60rem;

  @media only screen and (max-width: ${breakpoints.bpMobileL}) {
    width: 100%;
  }

  justify-content: center;
  align-items: center;

  margin-top: 8rem;
`;

const Title = styled.h4`
  font-weight: 500;
  align-self: flex-start;

  letter-spacing: 1px;
  margin: 0;
  margin-bottom: 1rem;

  @media only screen and (max-width: ${breakpoints.bpMobileL}) {
    margin-left: 1rem;
  }

  color: ${props => props.theme.fontColors.textRgba(0.8)};
`;

const StyledList = styled(List)`
  min-width: 100%;
  background-color: ${props => props.theme.panelBackgroundColor};

  &:not(:last-child) {
    margin-bottom: 3rem;
  }
`;

const Name = styled.span`
  color: ${props => props.theme.fontColors.textRgba(0.8)};
`;

const Description = styled.span`
  color: ${props => props.theme.fontColors.textRgba(0.6)};
`;

const mapStateToProps = (state: ApplicationState) => ({
  config: state.reducer.user.configuration
});

export default connect(
  mapStateToProps,
  null
)(Preferences);
