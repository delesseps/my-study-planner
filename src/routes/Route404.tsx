import * as React from "react";
import { ReactComponent as Logo } from "assets/logo.svg";
import styled from "styled-components";
import { Section404 } from "components";

interface I404Props {
  white?: boolean;
}

const Window404: React.FunctionComponent<I404Props> = ({ white }) => {
  return (
    <Wrapper>
      <LogoBox>
        <StyledLogo />
        <LogoTitle>My Study Planner</LogoTitle>
      </LogoBox>
      <Section404 white={white} />
    </Wrapper>
  );
};

const Wrapper = styled.section`
  display: flex;
  flex-direction: column;
  min-height: 100vh;
`;

const LogoBox = styled.div`
  display: flex;
  align-items: center;

  padding: 2rem;

  background-color: ${props => props.theme.backgroundColor};
`;

const StyledLogo = styled(Logo)`
  width: 5rem;
  height: 5rem;

  margin-right: 1.5rem;
`;

const LogoTitle = styled.h2`
  margin-bottom: 0.8rem;
  font-weight: 400;
  letter-spacing: 1px;

  color: ${props => props.theme.fontColors.text};
`;

export default Window404;
