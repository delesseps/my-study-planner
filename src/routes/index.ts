export { default as ChangePassword } from "./ChangePassword";
export { default as ForgotPassword } from "./ForgotPassword";
export { default as FriendsClasses } from "./FriendsClasses";
export { default as Grades } from "./Grades";
export { default as Home } from "./Home";
export { default as LinkGoogleAccount } from "./LinkGoogleAccount";
export { default as Preferences } from "./Preferences";
export { default as SignIn } from "./SignIn";
export { default as SignUp } from "./SignUp";
