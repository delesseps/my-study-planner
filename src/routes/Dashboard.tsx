import React, { useEffect } from "react";
import { FadeIn, Loading, Sidebar, TopBar } from "components";
import { connect, useDispatch } from "react-redux";
import { ApplicationState } from "store/types";
import { Switch, Route, Redirect } from "react-router";
import { breakpoints } from "styled";
import { requestUser, signOut } from "store/effects";
import { initializePush } from "firebase/initialize";
import IRequestError from "interfaces/IRequestError";
import IUser from "interfaces/IUser";
import styled from "styled-components";
import { Alert } from "antd";
import { Schedule } from "routes";

const Home = React.lazy(() => import("routes/Home"));
const Preferences = React.lazy(() => import("routes/Preferences"));

interface IDashboardProps {
  error: IRequestError | undefined;
  user: IUser;
}

const Dashboard: React.FC<IDashboardProps> = ({ error, user }) => {
  const dispatch = useDispatch();

  useEffect(() => {
    initializePush();
  }, []);

  useEffect(() => {
    dispatch(requestUser());
  }, [dispatch]);

  useEffect(() => {
    if (error && error.status === 401) dispatch(signOut());
  }, [error, dispatch]);

  return (
    <FadeIn>
      <Wrapper>
        <Sider>
          <Sidebar />
        </Sider>
        <Content>
          {!user.verified && (
            <EmailVerificationError
              message="Check your email and activate your account!"
              banner
              closable
            />
          )}
          <TopBar />
          <React.Suspense fallback={<Loading />}>
            <Switch>
              <Route path="/dashboard" exact component={Home} />
              <Route
                path="/dashboard/Preferences"
                exact
                component={Preferences}
              />
              <Route path="/dashboard/schedule" exact component={Schedule} />
              <Redirect to="/dashboard" />
            </Switch>
          </React.Suspense>
        </Content>
      </Wrapper>
    </FadeIn>
  );
};

const Wrapper = styled.main`
  display: flex;

  @media only screen and (max-width: ${breakpoints.bpMedium}) {
    flex-direction: column;
  }

  background-color: ${props => props.theme.backgroundColor};
`;

const EmailVerificationError = styled(Alert)`
  position: absolute;
  top: 0;
  left: 0;

  width: 100%;
  z-index: 100;
`;

const Sider = styled.section`
  flex: 1;
  min-height: 100vh;

  @media only screen and (max-width: ${breakpoints.bpMedium}) {
    min-height: auto;
    width: 100%;
    z-index: 1000;

    position: fixed;
    bottom: 0;
  }
`;

const Content = styled.section`
  flex: 0 0 92%;
  padding: 4rem 6rem;

  position: relative;

  @media only screen and (max-width: ${breakpoints.bpLargest}) {
    padding: 4rem 4rem;
  }

  @media only screen and (max-width: ${breakpoints.bpMedium}) {
    flex: 1;
    min-height: 100vh;
  }

  @media only screen and (max-width: ${breakpoints.bpMobileL}) {
    padding: 4rem 1rem 9rem 1rem;
  }

  @media only screen and (max-width: ${breakpoints.bpMobileS}) {
    padding: 4rem 0.2rem 10rem 0.2rem;
  }
`;

const mapStateToProps = (state: ApplicationState) => ({
  user: state.reducer.user,
  error: state.reducer.error.user
});

export default connect(mapStateToProps)(Dashboard);
