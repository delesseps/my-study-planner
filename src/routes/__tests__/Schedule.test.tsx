import React from "react";
import { render, fireEvent } from "test-utils";
import { Schedule } from "routes";

test("renders correctly", () => {
  const { asFragment } = render(<Schedule />);
  expect(asFragment()).toMatchSnapshot();
});

test("opens modal on click", () => {
  const { getByTestId, getByText } = render(<Schedule />);

  fireEvent.click(getByTestId("mondayCard"));
  getByText("Monday Courses");

  fireEvent.click(getByTestId("tuesdayCard"));
  getByText("Tuesday Courses");

  fireEvent.click(getByTestId("wednesdayCard"));
  getByText("Wednesday Courses");
});
