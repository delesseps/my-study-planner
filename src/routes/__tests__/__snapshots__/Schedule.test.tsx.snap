// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding-top: 4rem;
}

.c1 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c1 > *:not(:last-child) {
  margin-bottom: 2rem;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  font-weight: 600;
  font-size: 2.6rem;
  color: rgba(39,39,39,0.8);
}

.c4 {
  padding: 2rem 6rem;
  margin-left: 4rem;
  cursor: pointer;
  background-color: #fff;
  box-shadow: 0px 4px 12px rgba(0,0,0,0.1);
  border-radius: 4px;
}

<section
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <h1
          class="c3"
        >
          Monday
        </h1>
        <div
          class="c4"
          data-testid="mondayCard"
        >
          <div
            class="ant-empty"
          >
            <div
              class="ant-empty-image"
            >
              <svg
                width="80"
              >
                education-books.svg
              </svg>
            </div>
            <p
              class="ant-empty-description"
            >
              Click here to add new courses
            </p>
          </div>
        </div>
      </div>
      <div
        class="c2"
      >
        <h1
          class="c3"
        >
          Tuesday
        </h1>
        <div
          class="c4"
          data-testid="tuesdayCard"
        >
          <div
            class="ant-empty"
          >
            <div
              class="ant-empty-image"
            >
              <svg
                width="80"
              >
                education-books.svg
              </svg>
            </div>
            <p
              class="ant-empty-description"
            >
              Click here to add new courses
            </p>
          </div>
        </div>
      </div>
      <div
        class="c2"
      >
        <h1
          class="c3"
        >
          Wednesday
        </h1>
        <div
          class="c4"
          data-testid="wednesdayCard"
        >
          <div
            class="ant-empty"
          >
            <div
              class="ant-empty-image"
            >
              <svg
                width="80"
              >
                education-books.svg
              </svg>
            </div>
            <p
              class="ant-empty-description"
            >
              Click here to add new courses
            </p>
          </div>
        </div>
      </div>
      <div
        class="c2"
      >
        <h1
          class="c3"
        >
          Thursday
        </h1>
        <div
          class="c4"
        >
          <div
            class="ant-empty"
          >
            <div
              class="ant-empty-image"
            >
              <svg
                width="80"
              >
                education-books.svg
              </svg>
            </div>
            <p
              class="ant-empty-description"
            >
              Click here to add new courses
            </p>
          </div>
        </div>
      </div>
      <div
        class="c2"
      >
        <h1
          class="c3"
        >
          Friday
        </h1>
        <div
          class="c4"
        >
          <div
            class="ant-empty"
          >
            <div
              class="ant-empty-image"
            >
              <svg
                width="80"
              >
                education-books.svg
              </svg>
            </div>
            <p
              class="ant-empty-description"
            >
              Click here to add new courses
            </p>
          </div>
        </div>
      </div>
      <div
        class="c2"
      >
        <h1
          class="c3"
        >
          Saturday
        </h1>
        <div
          class="c4"
        >
          <div
            class="ant-empty"
          >
            <div
              class="ant-empty-image"
            >
              <svg
                width="80"
              >
                education-books.svg
              </svg>
            </div>
            <p
              class="ant-empty-description"
            >
              Click here to add new courses
            </p>
          </div>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
